src/frontend/language_phase.nit

```diff
@@ -0,0 +1,121 @@
+# Basic support of internationalization through the generation of id-to-string tables
+module language_phase
+
+private import annotation
+private import parser_util
+import astbuilder
+import gettext
+
+redef class ToolContext
+	# Main phase of `language`
+	var language_phase: Phase = new LanguagePhase(self, null)
+end
+
+class LanguagePhase
+	super Phase
+
+	redef fun process_annotated_node(nmodule, nat) do
+		if not nat.name == "language" then return
+
+		if not nmodule isa AModuledecl then
+			toolcontext.error(nmodule.location, "Cannot use annotation language on something other than a module")
+			return
+		end
+
+		var domain = nmodule.n_name.n_id.text
+
+		if nat.n_args.length != 1 then
+			toolcontext.error(nat.location, "Error: wrong number of arguments for annotation language, expected 1, got {nat.n_args.length}")
```

Comment:
In fact, I do not understand why you evn need an agrument? Is not the point of the wole l10n thing to not hard-code languages in the source-file?